<launch>

        <!-- ARGUMENTS -->
            <!-- DEFAULT behavior, append bagfile:=kitti_XX to BAG in the command line-->
            <arg name="bagfile" default="kitti_00" />

            <arg name="bagfilewithpath" default="/media/limongi/Volume/KITTI_RAW_DATASET/VO/$(arg bagfile).bag"/>  <!-- Your bagfilewithpath here -->
            <arg name="camera" default="/kitti_stereo" />                                           <!-- The namespace where images are published -->
            <arg name="publish_rate" default="0.5" />
            <arg name="rviz" default="$(find road_layout_estimation)/rviz/visual_odometry_kitti_viso.rviz" />
            <param name="/use_sim_time" value="true"/>
            <arg name="map" default="$(find ira_open_street_map)/maps/$(arg bagfile).osm" />

            <include file="$(find ira_open_street_map)/launch/$(arg bagfile).launch" />

        <!-- STATIC TRANFORMS -->
            <!-- Static transform fixing Viso moving frame axis -->
                <node pkg="tf" type="static_transform_publisher" name="visual_odometry_camera_frame_to_visual_odometry_car_frame"
                        args="0 0 0 0 -1.570796 1.570796 visual_odometry_camera_frame visual_odometry_car_frame 50" />
                <!-- Transform from local_map (osm package) to road_layout_estimation fixed frame-->
                <node pkg="tf" type="static_transform_publisher" name="odom_to_visual_odometry_odom_x_forward"
                        args="0 0 0 -1.570796 0 -1.570796 visual_odometry_odom_x_forward odom 50" />
                <!-- Transform from local_map (osm package) to road_layout_estimation fixed frame -->
                <node pkg="tf" type="static_transform_publisher" name="local_map_to_odom"
                        args="0 0 0 0 0 0 local_map visual_odometry_odom_x_forward 50" />
			
        <!-- ROSBAG PLAY
            <node pkg="rosbag" type="play" name="rosbag" args="-clock $(arg bagfilewithpath) -r $(arg publish_rate) --delay=5"/> -->

        <!-- VISUAL ODOMETRY STUFF -->
            <!-- Run the ROS package stereo_image_proc -->
            <group ns="$(arg camera)" >
                    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc" />
            </group>
            <!-- Run the viso2_ros package -->
            <node pkg="viso2_ros" type="stereo_odometer" name="stereo_odometer" output="screen">
                    <rosparam param="base_link_frame_id">visual_odometry_camera_frame</rosparam>
                    <remap from="stereo" to="$(arg camera)"/>
                    <remap from="image" to="image_rect"/>
            </node>
            <!-- Run image stereo viewer + disparity
            <node pkg="image_view" type="stereo_view" name="stereo_view">
                    <remap from="stereo" to="$(arg camera)"/>
                    <remap from="image" to="image_rect"/>
            </node>	 -->

        <!-- OSM QUERY NODE -->
            <node
                    pkg="ira_open_street_map"
                    type="osm_query_node"
                    name="osm_query_node"
                    output="screen"
                    args="$(arg map)"
            />


	<!-- Launch ROAD LAYOUT ESTIMATION node -->
	<node
		pkg="road_layout_estimation" type="road_layout_estimation" name="road_layout_estimation"
		required="true"
		output="screen"
		args="/stereo_odometer/odometry"
	/>  

        <!-- VISUALIZATION STUFF -->
            <!-- Launch RViz -->
            <node pkg="rviz" type="rviz" name="rviz" respawn="true"
                    args="-d $(arg rviz)"
                    output="screen"
            />

	<!-- Launch RQT reconfigure 
        <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" respawn="true"/> -->

        <!-- Launch ROAD LANE DETECTOR
            <node
                    pkg="road_lane_detection" type="bev_lane_detection_node" name="bev_lane_detection_node"
                    required="true"
                    output="screen"
            /> -->
</launch>
